File: 1379-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree/1379-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */

class Solution {
    
    public final TreeNode getTargetCopy_01(final TreeNode node, final TreeNode target) {
        //base case.
        if(node.val == target.val) return node;
        
        //left Recursive call
        if(node.left != null) {
            TreeNode leftAns = getTargetCopy_01(node.left, target);
            if(leftAns.val == target.val) {
                return leftAns;
            }
        }
        //right Recursive call
        if(node.right != null) {
            TreeNode rightAns = getTargetCopy_01(node.right, target);
                if(rightAns.val == target.val) {
                return rightAns;
            }
        }
        
        return new TreeNode(-1);
    }
    
    public final TreeNode getTargetCopy(final TreeNode original, final TreeNode cloned, final TreeNode target) {
        TreeNode ans = getTargetCopy_01(cloned,target);
        return ans;
    }
}

File: 1379-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree/1379-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

class Solution {
public:
    
    TreeNode* getTargetCopy_01(TreeNode* node, TreeNode* target) {
        
        // if node.val == target.val
        if(node->val == target->val) return node;
        
        //Recursive call for left
        if(node->left != nullptr) {
            TreeNode* left = getTargetCopy_01(node->left,target);
            if(left->val == target->val) return left;
        }
        
        //Recursive call for right
        if(node->right != nullptr) {
            TreeNode* right = getTargetCopy_01(node->right,target);
            if(right->val == target->val) return right;
        }
        
        return new TreeNode(-1);
    }
    
    TreeNode* getTargetCopy(TreeNode* original, TreeNode* cloned, TreeNode* target) {
        TreeNode* ans = getTargetCopy_01(cloned,target);
        return ans;
    }
};